website/docs/opengl/1-hello-window.md

```diff
@@ -0,0 +1,263 @@
+
+```cs
+WindowOptions options = WindowOptions.Default;
+```
+
+We'd still like to change a few of these options for our application, though, so go ahead and add the following underneath that line:
+
+```cs
+options.Size = new Vector2D<int>(800, 600);
+options.Title = "My first Silk.NET application!";
+```
+
+What these do should be fairly obvious but we'll go over them anyway:
+* **Size** - The initial size in pixels (aka resolution) of the window.
+* **Title** - The initial title (the text at the top) of the window.
+
+### Creating the window
+In order for windows to work properly, we need to keep a **reference** to our window after we've created it. This is in the form of an `IWindow`.
+
+Add the following to the top of your class:
+
+```cs
+private static IWindow _window;
+```
+
+Next, we need to create the window itself. This is done with the `Window.Create()` method. All we need to do is pass in our options and it will create a window for us.
+
+In your `Main` method, add the following underneath your window options code:
+
+```cs
+_window = Window.Create(options);
+```
+
+Now, you may think we have got enough code to run a window, but if you run the program you'll see nothing will show up. That's because we're missing one more thing.
+
+The last thing we need to do is to tell the window to run. At the bottom of your `Main` method, add the following:
+
+```cs
+_window.Run();
+```
+
+And that's it! Run the program and you should hopefully see a window.
+
+![Window](../../images/opengl/chapter1/window1.png)
+
+## Window Events
+The window we've just created has several events we can subscribe to, such as loading and rendering, which are key for your program to work properly.
+
+For this tutorial, we'll be subscribing to three events:
+* Load
+* Update
+* Render
+
+Add the following three methods to your class:
+
+```cs
+private static void OnLoad() { }
+
+private static void OnUpdate(double dt) { }
+
+private static void OnRender(double dt) { }
+```
+
+These are the methods that will power our main application. For this tutorial, we'll only be using `OnLoad`, but it's helpful to have all three.
+
+> [!Note]
+> The `dt` parameter (which you will often see labelled as `obj`), is the **delta time**, which is the amount of time, in seconds, that has passed since the last frame. Using delta time is key for framerate-independent actions, such as movement.
+
+Next, we need to subscribe to these events. Add the following code to your `Main` method, just after you call `Window.Create()`:
+
+```cs
+_window.Load += OnLoad;
+_window.Update += OnUpdate;
+_window.Render += OnRender;
+```
+
+Launch your program again and you will see that... nothing has changed. Good! That means it's working correctly. Try adding some logs in the load, update, and render methods to see exactly when they are called.
+
+![Loading, Updating & Rendering](../../images/opengl/chapter1/loading-rendering.png)
+
+We're now ready to handle some input!
```

Comment:
I would consider specifically mentioning/showing you added Console.WriteLine() (or similar) and say "Try temporarily adding [...] and render methods so we can see in the console when they are being called:"

Or in the "final result" code block at the end, add the logging too

Reply:
I've written that here: `Try adding some logs in the load, update, and render methods to see exactly when they are called.`, on line 161. Let me know if I should change it/make it clearer though. I've also added the logging to the final result now.

Reply:
(Just to be clear, I don't think the logging should stay in future tutorials, I just want to make sure newbies don't get confused and think that Silk is supposed to be printing that on its own, so it helps to be able to see that it was _you_ that added the code to do that)